### PR TITLE
Add ability to ping target, as well as running job statuses.

### PR DESCRIPTION
**Pull request by kbishop711**
Added functionality and some unit tests.

### CODE REVIEW

**Comment by kbishop711** [comment]
Not very sure how to properly test this. Any suggestions would be great.

File: rift/api/resources.py [hunk 1]
```diff
@@ -154,6 +158,57 @@ def on_delete(self, req, resp, tenant_id, target_id):
         Target.delete_target(target_id=target_id)
 
 
+class PingTargetResource(ApiResource):
+
+    def on_get(self, req, resp, tenant_id, target_id):
+        target = Target.get_target(tenant_id, target_id)
+        if target:
+            # Nova
+            if 'nova' in target.address.as_dict().keys():
```
Comment (jmvrbanac) [inline rift/api/resources.py]: You use target.address.* quite a bit, so I would assign that to a local variable, to keep from deep referencing a lot.

File: rift/api/resources.py [hunk 2]
```diff
@@ -154,6 +158,57 @@ def on_delete(self, req, resp, tenant_id, target_id):
...
+                creds = SSHKeyCredentials(
+                    username=ssh.get('username'),
+                    key_contents=ssh.get('private_key')
+                )
+                client = SSHClient(
+                    host=ip.address,
+                    port=ip.port,
+                    credentials=creds
+                )
+                print "still here"
```
Comment (jmvrbanac) [inline rift/api/resources.py]: We should be using a logger instead of print statements

File: spec/rift/api/resources/targets.py [hunk 3]
```diff
@@ -27,6 +27,25 @@ def can_get_target(self):
...
+        post_resp = self.app.post_json(
+            '/v1/tenant/targets',
+            VALID_TARGETS['hostname_target_with_ssh_password_auth']['body'])
+        expect(post_resp.status_int).to.equal(201)
+        expect(post_resp.json).to.contain('target_id')
+        target_id = post_resp.json['target_id']
+
+        resp = self.app.get('/v1/tenant/targets/{0}/ping'.format(target_id),
+                            expect_errors=True)
+        require(resp.status_int).to.be_in([200, 404])
```
Comment (jmvrbanac) [inline spec/rift/api/resources/targets.py]: 200 and 404? Shouldn't it be a 200 or a 404?

**Comment by jmvrbanac** [comment]
As far as testing, you should be able to patch into the get_driver function and get it to raise an exception to cover those lines.

**Comment by pglass** [comment]
Yeah, I patched `get_driver` here as an example: https://github.com/CloudRift/Rift/blob/master/spec/rift/plugins/nova.py

**Pull request merged**